payments-core/src/main/java/com/stripe/android/analytics/DefaultPaymentSessionEventReporter.kt

```diff
@@ -49,6 +49,7 @@ internal class DefaultPaymentSessionEventReporter @Inject internal constructor(
     }
 
     override fun onFormShown(code: PaymentMethodCode) {
+        durationProvider.end(DurationProvider.Key.ConfirmButtonClicked)
         durationProvider.start(DurationProvider.Key.ConfirmButtonClicked)
```

Comment:
Why is this necessary?

Reply:
`DurationProvider` by default will not reset a timer if `start` is called again. This can be problematic if the user enters the `AddNewPaymentMethod` screen, go backs and then enters into the `AddPaymentMethod` screen again. The timer isn't reset even though the event was meant to be called twice with a new start time, showing an invalid `duration` amount that started the first time `AddPaymentMethod` was shown and ended on the last time the ``AddPaymentMethod` screen was shown after the `confirm` button was clicked.

Reply:
I could update `DurationProvider` to have a default property in `start` to reset the timer

```kotlin
interface DurationProvider {
    fun start(key: DurationProvider.Key, reset: Boolean = false)
    fun end(key: DurationProvider.Key): Duration?
}
 ```

Reply:
I think we can add that. Right now we don't have `DurationProvider` process death handling. This is probably something we can look at at future date.